src/index.ts

```diff
@@ -142,8 +145,8 @@ export class Client {
     return this.connection().execute<T>(query, args, options)
   }
 
-  connection(): Connection {
-    return new Connection(this.config)
+  connection(override?: Config): Connection {
```

Comment:
I see why this was done out of convenience, but this leads to some behavior I don't think we'd want?

I feel like we'd want some type that covers just the database/variables command and not all of Config.